### ISSUE TITLE
Error: Getting the NetworkManager service state failed

### ISSUE THREAD

**Issue by tonyho** [resin-io/resin-wifi-connect#220]
I download the binary/ELF and run the following version files at my ARM linux, but it is not working.
Used version: wifi-connect-v4.1.1-linux-armv7hf.tar.gz

The relative log is: 
```
# ./wifi-connect 
Error: Getting the NetworkManager service state failed
  caused by: Service
# NetworkManager --version
1.10.2
# ps ax | grep Network
  260 root     /usr/sbin/NetworkManager
  528 root     grep Network
# 
```

My board is:
arm v7l Linux 4.1.15.  And the networkmanager is listening the port:
`
unix  2      [ ACC ]     STREAM     LISTENING       3344 /var/run/NetworkManager/private-dhcp

`

Also I can use the nmcli to control the Networkmanger.

**Comment by majorz** [comment]
Are you running wifi-connect in a resin OS container or another distro? Asking because this error usually occurs when running in a resin OS container, where `DBUS_SYSTEM_BUS_ADDRESS=unix:path=/host/run/dbus/system_bus_socket` should be exported before running the application.

**Comment by majorz** [comment]
The root cause is that wifi-connect does not find the NetworkManager service through the systemd D-Bus interface. Is it a systemd enabled system?

**Comment by majorz** [comment]
If so, probably we need to put a warning there and continue operation without exiting with an error.

**Comment by tonyho** [comment]
Thanks majorz. Our system is running with SysVInit init script instead of systemd, also, not the resin OS container, we build the rootfs using the Yocto.

About the D-Bus, I guess it's exposed for the nmcli and nmtui can interact with the NetworkManager. 

So, can this wifi-connect work on such system?  Or what should I do to make it work on the system?

**Comment by majorz** [comment]
Alright, so this is what I expected. Currently wifi-connect expects systemd to be active. I will fix this in the next release, so that you may use it under SysVInit. I should be able to get it done today.

**Comment by majorz** [comment]
This is now fixed in v4.1.2.
